src/Editor/ImGui/ImGuizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+        SCALE_X,
+        SCALE_Y,
+        SCALE_Z,
+        SCALE_XYZ,
+        BOUNDS
+    };
+
+    struct Context
+    {
+        Context() : mbUsing(false), mbEnable(true), mbUsingBounds(false)
+        {
+        }
+
+        ImDrawList* mDrawList;
+
+        MODE mMode;
+        matrix_t mViewMat;
+        matrix_t mProjectionMat;
+        matrix_t mModel;
+        matrix_t mModelInverse;
+        matrix_t mModelSource;
+        matrix_t mModelSourceInverse;
+        matrix_t mMVP;
+        matrix_t mViewProjection;
+
+        vec_t mModelScaleOrigin;
+        vec_t mCameraEye;
+        vec_t mCameraRight;
+        vec_t mCameraDir;
+        vec_t mCameraUp;
+        vec_t mRayOrigin;
+        vec_t mRayVector;
+
+        ImVec2 mScreenSquareCenter;
+        ImVec2 mScreenSquareMin;
+        ImVec2 mScreenSquareMax;
+
+        float mScreenFactor;
+        vec_t mRelativeOrigin;
+
+        bool mbUsing;
+        bool mbEnable;
+
+        // translation
+        vec_t mTranslationPlan;
+        vec_t mTranslationPlanOrigin;
+        vec_t mMatrixOrigin;
+
+        // rotation
+        vec_t mRotationVectorSource;
+        float mRotationAngle;
+        float mRotationAngleOrigin;
+        //vec_t mWorldToLocalAxis;
+
+        // scale
+        vec_t mScale;
+        vec_t mScaleValueOrigin;
+        float mSaveMousePosx;
+
+        // save axis factor when using gizmo
+        bool mBelowAxisLimit[3];
+        bool mBelowPlaneLimit[3];
+        float mAxisFactor[3];
+
+        // bounds stretching
+        vec_t mBoundsPivot;
+        vec_t mBoundsAnchor;
+        vec_t mBoundsPlan;
+        vec_t mBoundsLocalPivot;
+        int mBoundsBestAxis;
+        int mBoundsAxis[2];
+        bool mbUsingBounds;
+        matrix_t mBoundsMatrix;
+
+        //
+        int mCurrentOperation;
+
+        float mX = 0.f;
+        float mY = 0.f;
+        float mWidth = 0.f;
+        float mHeight = 0.f;
+    };
+
+    static Context gContext;
+
+    static const float angleLimit = 0.96f;
+    static const float planeLimit = 0.2f;
+
+    static const vec_t directionUnary[3] = { makeVect(1.f, 0.f, 0.f), makeVect(0.f, 1.f, 0.f), makeVect(0.f, 0.f, 1.f) };
+    static const ImU32 directionColor[3] = { 0xFF0000AA, 0xFF00AA00, 0xFFAA0000 };
+    static const ImU32 selectionColor = 0xFF1080FF;
+    static const ImU32 inactiveColor = 0x99999999;
+    static const ImU32 translationLineColor = 0xAAAAAAAA;
+    static const char *translationInfoMask[] = { "X : %5.3f", "Y : %5.3f", "Z : %5.3f", "X : %5.3f Y : %5.3f", "Y : %5.3f Z : %5.3f", "X : %5.3f Z : %5.3f", "X : %5.3f Y : %5.3f Z : %5.3f" };
+    static const char *scaleInfoMask[] = { "X : %5.2f", "Y : %5.2f", "Z : %5.2f", "XYZ : %5.2f" };
+    static const char *rotationInfoMask[] = { "X : %5.2f deg %5.2f rad", "Y : %5.2f deg %5.2f rad", "Z : %5.2f deg %5.2f rad", "Screen : %5.2f deg %5.2f rad" };
+    static const int translationInfoIndex[] = { 0,0,0, 1,0,0, 2,0,0, 0,1,0, 1,2,0, 0,2,1, 0,1,2 };
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') comma should be followed by whitespace [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)